ddtrace/tracer/option.go

```diff
@@ -489,7 +518,21 @@ func newConfig(opts ...StartOption) (*config, error) {
 	// is set, but DD_TRACE_PARTIAL_FLUSH_ENABLED is not true. Or just assume it should be enabled
 	// if it's explicitly set, and don't require both variables to be configured.
 
-	c.dynamicInstrumentationEnabled, _, _ = stableconfig.Bool("DD_DYNAMIC_INSTRUMENTATION_ENABLED", false)
+	dynamicInstrumentationEnabledDefault, origin, _ := stableconfig.Bool("DD_DYNAMIC_INSTRUMENTATION_ENABLED", false)
+	c.dynamicInstrumentationEnabled = newDynamicConfig(
+		"dynamic_instrumentation_enabled",
+		dynamicInstrumentationEnabledDefault,
+		func(bool) bool {
+			// NOTE: the side effects of changes are performed in onRemoteConfigUpdate.
+			return true
+		}, /* apply */
+		equal[bool],
+	)
+	// Keep track of the origin of the config value. Note that the origin is not
+	// properly maintained when the config value changes, but that's not a problem
+	// for us since we won't allow the value to change if dynamic instrumentation
+	// is explicitly disabled.
```

Comment:
By this, do you just mean, "If the config value changes, the initiator of this change is responsible for modifying the origin accordingly"?

Reply:
More like I was trying to say that the deficiency of the `dynamicConfig` about not maintaining the origin properly doesn't affect our use of the origin for this particular setting (because in the case where the origin is relevant, we never mutate the setting).
But I've now removed the comment altogether...